app/ssttpcalculator/CalculatorController.scala

```diff
@@ -158,26 +158,38 @@ class CalculatorController @Inject() (
         journey.remainingIncomeAfterSpending
       )
 
-      val minCustomAmount = defaultPlanOptions.values
-        .headOption.fold(BigDecimal(1))(_.firstInstalment.amount)
-      val maxCustomAmount = paymentPlansService.maximumPossibleInstalmentAmount(journey)
-
-      val allPlanOptions = maybePreviousCustomAmount(journey, defaultPlanOptions) match {
-        case None                 => defaultPlanOptions
-        case Some(customSchedule) => Map((0, customSchedule)) ++ defaultPlanOptions
-      }
-
       if (defaultPlanOptions.isEmpty) {
         Redirect(ssttpaffordability.routes.AffordabilityController.getWeCannotAgreeYourPP())
       } else {
-        Ok(views.how_much_can_you_pay_each_month_form(
-          routes.CalculatorController.submitCalculateInstalments(),
-          selectPlanForm(minCustomAmount, maxCustomAmount),
-          allPlanOptions,
-          minCustomAmount.setScale(2, HALF_UP),
-          maxCustomAmount.setScale(2, HALF_UP),
-          journey.maybePlanSelection))
+        appConfig.calculatorType match {
+
+          case CalculatorType.Legacy =>
+            Ok(views.how_much_can_you_pay_each_month_form_legacy(
+              routes.CalculatorController.submitCalculateInstalments(),
+              selectPlanForm(),
+              defaultPlanOptions,
+              journey.maybePlanSelection))
+
+          case CalculatorType.PaymentOptimised =>
+            val minCustomAmount = defaultPlanOptions.values
```

Comment:
`defaultPlanOptions` has type `Map[Int, PaymentSchedule]`. From that type there is no guarantee that the head will give you the least amount - it would be safer to do a sort before doing head.

Also, from the above `defaultPlanOptions.isEmpty` check above you know that `headOption` will return you something so the `fold` with a default case that you never use can be avoided by maybe doing something like this:
```scala
defaultPlanOptions.values.sortBy(...).headOption match {
  case None =>
    Redirect(ssttpaffordability.routes.AffordabilityController.getWeCannotAgreeYourPP())

  case Some(minAmount) =>
    ...
}
```